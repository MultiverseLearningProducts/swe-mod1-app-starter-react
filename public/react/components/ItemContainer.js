import React, { useState } from 'react';
import { Item } from './Item';

export const ItemContainer = (props) => {
	const {onAdd} = props; // props.onAdd
	const {onView}=props;

	return (
		<div className="item-container">
		{
			props.items.map((item, idx) => {
				return <Item onAdd={onAdd} onView={} item={item} key={idx} />;
			
			})

		}
		</div>
	)
} 
