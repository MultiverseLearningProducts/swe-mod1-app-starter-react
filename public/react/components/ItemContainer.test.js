import React from 'react';
import renderer from 'react-test-renderer';

import { ItemContainer } from './ItemContainer';

describe("Item container component", () => {
  test("Contains items passed down as props", () => {
    const sauces = ['sriracha', 'franks'];
    const component = renderer.create(<ItemContainer items={sauces}/>);
    const treeEl = component.toTree()

    expect(treeEl.props.items.length).toBe(2)
  })
})