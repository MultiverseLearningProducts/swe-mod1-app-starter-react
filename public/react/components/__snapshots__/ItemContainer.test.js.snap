// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item container component renders Item Container component as expected 1`] = `
<div
  className="item-container"
>
  <div
    className="card"
  >
    <div
      className="hover-container"
    >
      <img
        alt="Fan Flower"
        className="card-img"
        src="https://cdn.shopify.com/s/files/1/0317/0687/3992/products/191-large_default_bdf68ca7-5284-412c-9fec-78235d9cd046.jpg?v=1580438028"
      />
      <div
        className="overlay"
      >
        <button
          aria-label="Favorite"
          className="cart-btn"
          onClick={[Function]}
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
              }
            }
            data-emotion="css i4bv87-MuiSvgIcon-root"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cart-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="FavoriteBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
        </button>
        <a
          aria-label="View More Details"
          className="cart-btn"
          href="/plant/undefined"
          onClick={[Function]}
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
              }
            }
            data-emotion="css i4bv87-MuiSvgIcon-root"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cart-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </a>
        <button
          aria-label="Add To Cart"
          className="cart-btn"
          onClick={[Function]}
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
              }
            }
            data-emotion="css i4bv87-MuiSvgIcon-root"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cart-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="ShoppingCartOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </button>
      </div>
    </div>
    <h3
      className="card-name"
    >
      Fan Flower
       
    </h3>
    <h3
      className="card-price"
    >
      $ 
      12.99
    </h3>
  </div>
</div>
`;
